mod/primitives/pkg/encoding/hex/format.go

```diff
@@ -20,14 +20,11 @@
 
 package hex
 
-// has0xPrefix returns true if s has a 0x prefix.
-func has0xPrefix[T []byte | string](s T) bool {
-	return len(s) >= 2 && s[0] == '0' && (s[1] == 'x' || s[1] == 'X')
-}
```

Comment:
absorbed into `IsValidHex` to consolidate verifications (which makes them less error prone)